### ISSUE TITLE
Virtual network without any subnets is invalid

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#1303]
**Description of the issue**

A VNET without any subnets fails the following rules `Azure.VNET.UseNSGs` and `Azure.VNET.SubnetName`.

**Expected behaviour**

A virtual network without any subnets should pass these rules.

**Error output**

The following reasons are reported:

```text
- The subnet () has no NSG associated.
```

```text
- The parameter 'inputObject' is null.
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.13.3**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.2.1
PSEdition                      Core
GitCommitId                    7.2.1
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
